tools/acorn-optimizer.js

```diff
@@ -53,13 +54,15 @@ function visitChildren(node, c) {
     }
     return false;
   }
-  for (var key in node) {
-    var child = node[key];
+  for (const key in node) {
     // Check for a child.
-    if (!maybeChild(child)) {
-      // Check for an array of children.
-      if (Array.isArray(child)) {
-        child.forEach(maybeChild);
+    if (Object.prototype.hasOwnProperty.call(node, key)) {
```

Comment:
Does lint auto-change code in LHS to RHS by adding these lines? (Not very familiar with eslint myself)

Reply:
No, the linter complains that when you iterate through an object with `for in` you need to include an inner check for hasOwnProperty:

Here is the rule: https://eslint.org/docs/rules/guard-for-in